boa/src/builtins/math/mod.rs

```diff
@@ -165,6 +166,25 @@ impl Math {
         Ok(args.get(0).map_or(f64::NAN, |x| f64::from(x).ceil()).into())
     }
 
+    /// Get the number of leading zeros in the 32 bit representation of a number
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///  - [MDN documentation][mdn]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-math.clz32
+    /// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/clz32
+    pub(crate) fn clz32(_: &Value, args: &[Value], _: &mut Interpreter) -> ResultValue {
+        Ok(args
+            .get(0)
+            .and_then(|x| f64::from(x).to_u32())
+            .map_or(32usize, |u| {
+                let s = format!("{:032b}", u);
+                s.chars().take_while(|x| *x == '0').count()
+            })
```

Comment:
```suggestion
            .and_then(|x| i32::from(x).leading_zeros())
            .unwrap_or(32)
```
A better way of doing this would we to use [`.leading_zeros()`](https://doc.rust-lang.org/std/primitive.i32.html#method.leading_zeros) which would eliminate the allocation from `format!`